File: src/main/java/mods/cybercat/gigeresque/client/entity/render/RunnerbursterEntityRenderer.java
package mods.cybercat.gigeresque.client.entity.render;

import com.mojang.blaze3d.vertex.PoseStack;
import com.mojang.blaze3d.vertex.VertexConsumer;

import mod.azure.azurelib.cache.object.BakedGeoModel;
import mod.azure.azurelib.renderer.GeoEntityRenderer;
import mods.cybercat.gigeresque.client.entity.model.RunnerbursterEntityModel;
import mods.cybercat.gigeresque.client.entity.render.feature.RunnerBusterBloodFeatureRenderer;
import mods.cybercat.gigeresque.common.entity.impl.RunnerbursterEntity;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.minecraft.client.renderer.MultiBufferSource;
import net.minecraft.client.renderer.entity.EntityRendererProvider;

@Environment(EnvType.CLIENT)
public class RunnerbursterEntityRenderer extends GeoEntityRenderer<RunnerbursterEntity> {
	public RunnerbursterEntityRenderer(EntityRendererProvider.Context context) {
		super(context, new RunnerbursterEntityModel());
		this.shadowRadius = 0.3f;
		this.addRenderLayer(new RunnerBusterBloodFeatureRenderer(this));
	}

	@Override
	public void preRender(PoseStack poseStack, RunnerbursterEntity animatable, BakedGeoModel model, MultiBufferSource bufferSource, VertexConsumer buffer, boolean isReRender, float partialTick, int packedLight, int packedOverlay, float red, float green, float blue, float alpha) {
		float scaleFactor = 1.0f + (animatable.getGrowth() / animatable.getMaxGrowth());
		poseStack.pushPose();
		poseStack.scale(scaleFactor, scaleFactor, scaleFactor);
		poseStack.popPose();
		super.preRender(poseStack, animatable, model, bufferSource, buffer, isReRender, partialTick, packedLight, packedOverlay, red, green, blue, alpha);
	}

	@Override
	protected float getDeathMaxRotation(RunnerbursterEntity entityLivingBaseIn) {
		return 0;
	}
}

File: src/main/kotlin/com/bvanseg/gigeresque/client/entity/render/RunnerbursterEntityRenderer.kt
//package com.bvanseg.gigeresque.client.entity.render
//
//import com.bvanseg.gigeresque.client.entity.model.RunnerbursterEntityModel
//import com.bvanseg.gigeresque.client.extensions.scale
//import com.bvanseg.gigeresque.common.entity.impl.RunnerbursterEntity
//import net.fabricmc.api.EnvType
//import net.fabricmc.api.Environment
//import net.minecraft.client.render.VertexConsumerProvider
//import net.minecraft.client.render.entity.EntityRendererFactory
//import net.minecraft.client.util.math.MatrixStack
//import software.bernie.geckolib3.renderers.geo.GeoEntityRenderer
//
///**
// * @author Boston Vanseghi
// */
//@Environment(EnvType.CLIENT)
//class RunnerbursterEntityRenderer(context: EntityRendererFactory.Context) : GeoEntityRenderer<RunnerbursterEntity>(
//    context,
//    RunnerbursterEntityModel()
//) {
//    init {
//        this.shadowRadius = 0.3f
//    }
//
//    override fun render(
//        entity: RunnerbursterEntity,
//        entityYaw: Float,
//        partialTicks: Float,
//        stack: MatrixStack,
//        bufferIn: VertexConsumerProvider,
//        packedLightIn: Int
//    ) {
//        stack.scale(1.0f + (entity.growth / entity.maxGrowth.toFloat()))
//        super.render(entity, entityYaw, partialTicks, stack, bufferIn, packedLightIn)
//    }
//}